test/integration/src/test/java/io/pravega/test/integration/KVTableWithStreamTest.java

```diff
@@ -0,0 +1,769 @@
+        log.info("Deleting stream {}", streamName);
+        assertTrue(deleteStreamStatus.get());
+        // delete the  scope.
+        CompletableFuture<Boolean> deleteScopeStatus = controller.deleteScope(newScope);
+        log.info("Deleting scope {}", newScope);
+        assertTrue(deleteScopeStatus.get());
+
+        log.info("Successfully completed creating Stream & KVTable with same " +
+                "name under same scope with event read write test on both entities");
+    }
+
+    /**
+     * Tests Creation and deletion of KVTables and Streams +
+     * under different scopes
+     */
+    @Test
+    public void testDeleteKVTableStream() throws ExecutionException, InterruptedException {
+        String newScope1 = "TestScope1-" + randomAlphanumeric(5);
+        String newScope2 = "TestScope2-" + randomAlphanumeric(5);
+        KeyValueTableInfo newKVT1 = newKeyValueTableName(newScope1);
+        KeyValueTableInfo newKVT2 = newKeyValueTableName(newScope2);
+        KeyValueTableConfiguration config = KeyValueTableConfiguration.builder().partitionCount(5).build();
+        String streamName1 = newKVT1.getKeyValueTableName();
+        String streamName2 = newKVT2.getKeyValueTableName();
+        String readerGroupName = "readerGroup";
+        String readerName = "reader";
+        String routingKey = "routingKey";
+        String streamEvent = "Stream event";
+        int numEvents = 100;
+        int writeCount = 0;
+        int readCount = 0;
+
+        assertTrue("The requested Scope must get created", this.controller.createScope(newScope1).join());
+        // Creating KVT1 under Scope1.
+        assertTrue("The requested KVT must get created",
+                this.controller.createKeyValueTable(newScope1, newKVT1.getKeyValueTableName(),
+                        config).join());
+
+        assertEquals("Table partition value should be same as passed", DEFAULT_CONFIG.getPartitionCount(),
+                Futures.getAndHandleExceptions(this.controller.getCurrentSegmentsForKeyValueTable(newScope1,
+                        newKVT1.getKeyValueTableName()), RuntimeException::new).getSegments().size());
+
+        val tableSegments1 = this.controller.getCurrentSegmentsForKeyValueTable(newScope1,
+                newKVT1.getKeyValueTableName()).join();
+
+        for (val s : tableSegments1.getSegments()) {
+            // We know there's nothing in these segments. But if the segments hadn't been created, then this will throw
+            // an exception.
+            this.tableStore.get(s.getKVTScopedName(), Collections.singletonList(new ByteArraySegment(new byte[1])), TIMEOUT).join();
+        }
+
+        // Deleting KeyValueTable where Scope does not exist.
+        ResultObserver<DeleteKVTableStatus> result1 = new ResultObserver<>();
+        this.controllerService.deleteKeyValueTable(ModelHelper.createKeyValueTableInfo(newScope2,
```

Comment:
Is this an async call? Should you wait?

Reply:
Same below

Reply:
No, not an async call just trying to check the deletion status in case where a KVT does not exists in the requested scope.

Reply:
`deleteKeyValueTable` returns a CompletableFuture. See my other comment about async calls.

Reply:
@andreipaduroiu It is not the same `deleteKeyValueTable` api which we often use rather it is this one `public void deleteKeyValueTable(KeyValueTableInfo request, StreamObserver<DeleteKVTableStatus> responseObserver)` which does not returns anything as there is already a `responseObserver` to check for the completion of request from return status through enums. Thanks